### ISSUE TITLE
error: command 'intltool-update' failed: No such file or directory

### ISSUE THREAD

**Issue by arisboch** [jeromerobert/pdfarranger#81]
I'm trying to build PDFArranger under Windows, but I ran into various problems: some dependencies were present in msys (is that a user or a repository? I'm just totally new at using MSYS2), but not in mingw64, so I had to install that way (e.g. "mingw-w64-x86_64-python3-distutils-extra" instead of just "python3-distutils-extra"), but then I run into a problem: I got the error message `error: command 'intltool-update' failed: No such file or directory` and there is no "mingw64" version present, so how can I proceed here?

**Comment by arisboch** [comment]
I changed to username "msys" an it solved the problem with intltool, but a new problem sprung up: When I try to execute the command ``/mingw64/bin/python3 setup_win32.py bdist_msi`, I get the error message(s)

```
running bdist_msi
running build
running build_exe
Traceback (most recent call last):
  File "setup_win32.py", line 140, in <module>
    shortcutDir='StartMenuFolder'
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\dist.py", line 349, in setup
    distutils.core.setup(**attrs)
  File "C:/msys64/mingw64/lib/python3.7\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:/msys64/mingw64/lib/python3.7\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:/msys64/mingw64/lib/python3.7\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\windist.py", line 365, in run
    self.run_command('build')
  File "C:/msys64/mingw64/lib/python3.7\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:/msys64/mingw64/lib/python3.7\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:/msys64/mingw64/lib/python3.7\distutils\command\build.py", line 135, in run
    self.run_command(cmd_name)
  File "C:/msys64/mingw64/lib/python3.7\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "C:/msys64/mingw64/lib/python3.7\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\dist.py", line 218, in run
    zipExcludePackages = self.zip_exclude_packages)
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\freezer.py", line 150, in __init__
    self._VerifyConfiguration()
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\freezer.py", line 470, in _VerifyConfiguration
    sourceFileName)
cx_Freeze.freezer.ConfigError: cannot find file/directory named C:/msys64/mingw64/share/icons/Adwaita/16x16/actions/pan-down-symbolic.symbolic.png
```
and when I tried `/mingw64/bin/python3 setup_win32.py bdist_zip`, I got something similar:

```
running bdist_zip
Traceback (most recent call last):
  File "setup_win32.py", line 140, in <module>
    shortcutDir='StartMenuFolder'
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\dist.py", line 349, in setup
    distutils.core.setup(**attrs)
  File "C:/msys64/mingw64/lib/python3.7\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:/msys64/mingw64/lib/python3.7\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:/msys64/mingw64/lib/python3.7\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup_win32.py", line 124, in run
    build_exe.run()
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\dist.py", line 218, in run
    zipExcludePackages = self.zip_exclude_packages)
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\freezer.py", line 150, in __init__
    self._VerifyConfiguration()
  File "C:/msys64/mingw64/lib/python3.7/site-packages\cx_Freeze\freezer.py", line 470, in _VerifyConfiguration
    sourceFileName)
cx_Freeze.freezer.ConfigError: cannot find file/directory named C:/msys64/mingw64/share/icons/Adwaita/16x16/actions/pan-down-symbolic.symbolic.png
```

I *did* edit the freezer.py file as the Win32.md file said and instaling the packages "mingw-w64-i686-adwaita-icon-theme" and "mingw-w64-x86_64-adwaita-icon-theme" didn't help, either.

**Comment by jeromerobert** [comment]
> I'm trying to build PDFArranger under Windows

Good, you are probably the first one after me, so it'll help to robustify the process.

> I got the error message error: command 'intltool-update' failed: No such file or directory

`mingw-w64-x86_64-python3-distutils-extra` won't work because distutils-extra need `intltool-update` which is perl, and perl is not available in mingw. That's why you must first run `setup.py build` must be run with a MSYS2 python and not a mingw64 python.

> `cx_Freeze.freezer.ConfigError: cannot find file/directory named C:/msys64/mingw64/share/icons/Adwaita/16x16/actions/pan-down-symbolic.symbolic.png`

Is this file available on your system ?

> I did edit the freezer.py file as the Win32.md file said

You have to, but it's not related to the current issue

> the packages `mingw-w64-i686-adwaita-icon-theme` and `mingw-w64-x86_64-adwaita-icon-theme` didn't help, either

`mingw-w64-x86_64-adwaita-icon-theme` is needed but it's a reversed dependency of `mingw-w64-x86_64-gtk3` (see `pacman -Qi mingw-w64-x86_64-adwaita-icon-theme`). You should not have to install it explicitly.

**Comment by arisboch** [comment]
>mingw-w64-x86_64-python3-distutils-extra won't work because distutils-extra need intltool-update which is perl, and perl is not available in mingw. That's why you must first run setup.py build must be run with a MSYS2 python and not a mingw64 python.

I tried, but it didn't help ;-(

>Is this file available on your system ?

No.

**Comment by jeromerobert** [comment]
What does `pacman -Ql mingw-w64-x86_64-adwaita-icon-theme | grep pan-down` returns ?

**Comment by arisboch** [comment]
The following:

```
$ pacman -Ql mingw-w64-x86_64-adwaita-icon-theme | grep pan-down
mingw-w64-x86_64-adwaita-icon-theme /mingw64/share/icons/Adwaita/16x16/ui/pan-down-symbolic.symbolic.png
mingw-w64-x86_64-adwaita-icon-theme /mingw64/share/icons/Adwaita/24x24/ui/pan-down-symbolic.symbolic.png
mingw-w64-x86_64-adwaita-icon-theme /mingw64/share/icons/Adwaita/32x32/ui/pan-down-symbolic.symbolic.png
mingw-w64-x86_64-adwaita-icon-theme /mingw64/share/icons/Adwaita/48x48/ui/pan-down-symbolic.symbolic.png
mingw-w64-x86_64-adwaita-icon-theme /mingw64/share/icons/Adwaita/64x64/ui/pan-down-symbolic.symbolic.png
mingw-w64-x86_64-adwaita-icon-theme /mingw64/share/icons/Adwaita/96x96/ui/pan-down-symbolic.symbolic.png
mingw-w64-x86_64-adwaita-icon-theme /mingw64/share/icons/Adwaita/scalable/ui/pan-down-symbolic.svg
```

**Comment by jeromerobert** [comment]
Sorry but I have no idea why pacman says it installed `16x16/ui/pan-down-symbolic.symbolic.png` and why you don't have that file.

**Comment by arisboch** [comment]
When I'm back home I'm gonna reinstall MSYS2, maybe that'll help.

**Comment by arisboch** [comment]
I tried and it didn't work ;-(

**Comment by jeromerobert** [comment]
Closed by #83
